vit_jax/models.py

```diff
@@ -207,23 +208,37 @@ def apply(self,
             representation_size=None,
             classifier='gap'):
 
+    # (Possibly partial) ResNet root.
+    if resnet is not None:
+      width = int(64 * resnet.width_factor)
+
+      # Root block.
+      x = models_resnet.StdConv(
+          x, width, (7, 7), (2, 2), bias=False, name='conv_root')
+      x = nn.GroupNorm(x, name='gn_root')
+      x = nn.relu(x)
+      x = nn.max_pool(x, (3, 3), strides=(2, 2), padding='SAME')
+
+      # ResNet stages.
+      x = models_resnet.ResNetStage(
+          x, resnet.num_layers[0], width, first_stride=(1, 1), name='block1')
+      for i, block_size in enumerate(resnet.num_layers[1:], 1):
+        x = models_resnet.ResNetStage(
+            x,
+            block_size,
+            width * 2**i,
+            first_stride=(2, 2),
+            name=f'block{i + 1}')
+
     n, h, w, c = x.shape
 
-    # Embed the grid or patches of the grid.
-    fh, fw = patches.size
-    gh, gw = h // fh, w // fw
-    if hidden_size:  # We can merge s2d+emb into a single conv; it's the same.
```

Comment:
Is it expected that you removed the check for hidden_size here?

Reply:
We [always](https://github.com/google-research/vision_transformer/blob/master/vit_jax/configs.py) set `hidden_size` so I thought I would simply remove this dead code. Also IIRC the reshaping can lead to very bad padding no some architectures.